crates/forge_permission_resolver/src/permissions_cache.rs

```diff
@@ -0,0 +1,261 @@
+use regex::Regex;
+use std::collections::HashMap;
+use std::env;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use crate::permissions_resolver::PermissionHashMap;
+use crate::serde::{SerializablePermissionHashMap, ToStringMap};
+use std::fs::{self, File};
+use std::io::{Read, Write};
+use tracing::debug;
+
+const CACHE_EXPIRATION: Duration = Duration::from_secs(60 * 60 * 24 * 7); // 1 week
+
+fn default_cache_path() -> PathBuf {
+    if let Ok(home) = env::var("HOME") {
+        PathBuf::from(home).join(".cache/fsrt")
+    } else {
+        PathBuf::new()
+    }
+}
+
+/// Struct to hold cache-related settings
```

Comment:
Also, should the `Default` implementation just defer to `default_cache_path`?

Reply:
Also, `default_cache_path()` returns and empty `PathBuf` if there's no home directory. You could change that to return an `Option`, so that we don't try to do lookups when there's no home directory.